doc/api/n-api.md

```diff
@@ -2267,6 +2267,7 @@ typedef enum {
   napi_float64_array,
   napi_bigint64_array,
   napi_biguint64_array,
+  napi_float16_array
```

Comment:
Can you add a doc comment with a `changes:` section for the `napi_typedarray_type` header? You'll find other examples of that in this doc